pipeline/frontend/yaml/linter/linter.go

```diff
@@ -351,7 +383,7 @@ func (l *Linter) lintBadHabits(config *WorkflowConfig) (err error) {
 			if field != "" {
 				err = multierr.Append(err, &errorTypes.PipelineError{
 					Type:    errorTypes.PipelineErrorTypeBadHabit,
-					Message: "Please set an event filter on all when branches",
+					Message: "Please set an event filter for all steps/whole workflow on all when branches",
```

Comment:
#3630

1. What is the `when branches`? Is it ['when' block with 'branch' filter](https://woodpecker-ci.org/docs/2.3/usage/workflow-syntax#branch)?
2. If I don't have 'event' filter, but 50 steps, will I have 50 `bad_habit` warnings?
3. Should warning contain a link to the docs?
```
Please set an event filter for all [steps](https://woodpecker-ci.org/docs/2.3/usage/workflow-syntax#event) / whole [workflow](https://woodpecker-ci.org/docs/2.3/usage/workflow-syntax#event-1)
```

Reply:
> has two "branches": one for event = push and branch = main and one for event = tag.

I think the word `branches` for this is confusing as its mixed up with a git branch. 

So something as suggested by @zc-devs like `Please set a some filter for the [workflow](https://woodpecker-ci.org/docs/2.3/usage/workflow-syntax#event-1) or all [steps](https://woodpecker-ci.org/docs/2.3/usage/workflow-syntax#event).` might be better.

Reply:
> No - it's every "subfilter"

then maybe

> ... on all items in 'when' block.

if you want to have this end of sentence.

But short version with links is enough, I think.

Reply:
We could also start a new page in the docs similar to how eslint is explaining rules that describes the warning in more detail and gives positive and negative examples.